templates/userauth/account/all_user_chats.html

```diff
@@ -1,109 +1,129 @@
+                                                                                        name="action"
+                                                                                        value="remove_swimmer"
+                                                                                        class="button button-danger">
+                                                                                    Block
+                                                                                </button>
                                                                             </div>
                                                                         </div>
                                                                     </div>
                                                                 </div>
-                                                            </section>
-                                                        </div>
+                                                            </div>
+                                                        </section>
                                                     </div>
-                                                 </div>
-                                              </div>
-                                           </div>
-                                    {% endwith %}
-                                </div>
-                            </a>
-                        {% endwith %}
-                    {% endwith %}
-                {% endif %}
-                {% include "partials/vertical-spacer.html" with space="6" %}
-            {% endfor %}
+                                                </div>
+                                            </div>
+                                        </div>
+                                    </div>
+                                {% endwith %}
+                            </div>
+                        </a>
+                        {% include "partials/vertical-spacer.html" with space="6" %}
+                    {% endfor %}
+                </div>
+                <div class="grow overflow-y-auto">
+                    {% block user_chat %}
+                        {% comment %}
+                            When this block is not overriden, it means there
+                            is no user chat selected.
+
+                            On desktop we automatically then redirect them to
+                            the most recent chat.
+                        {% endcomment %}
+                        {% if first_user_chat %}
+                            {% user_chat_slug first_user_chat as user_slug %}
+                            {% url 'user_chat' user_slug as chat_url %}
+                            <div data-url="{{ chat_url }}"
+                                 x-init="if ($screen.isDesktop) { window.location = $el.dataset.url }"></div>
+                        {% endif %}
+                    {% endblock %}
```

Comment:
This is the slightly clunky method to go to the top chat when you load the page on desktop....

It would be nicer if the server would redirect to it when requesting the chat over page, but currently the server doesn't know whether it's desktop/mobile, and I don't think it should either.

A better solution might be to have the link to the messages page have a paramter like `latest=true` or something, which can be set clientside depending on `$screen.isDesktop`, and if the server sees that, it'll do the redirect.

Reply:
> A better solution might be to have the link to the messages page have a paramter like `latest=true` or something, which can be set clientside depending on `$screen.isDesktop`, and if the server sees that, it'll do the redirect.

Hmm yeah, sounds like a good approach, maybe in a future iteration.